client/src/Counter.js

```diff
@@ -0,0 +1,72 @@
+import { useState } from "react"
+const Counter = ({ setAllMyVideos, videoId, videoRating }) => {
+    const [rating, setRating] = useState(videoRating)
+    let newRating
+    const handleCounterDown = () => {
+        if (rating <= 0) {
+            newRating = 0
+            setRating(newRating)
+            handleRatingChange(newRating)
+            getAllNewVideos()
+        } else {
+            newRating = rating - 1
+            setRating(newRating)
+            handleRatingChange(newRating)
+            getAllNewVideos()
+        }
+    }
+    const handleCounterUp = () => {
+        newRating = rating + 1
+        setRating(newRating)
+        handleRatingChange(newRating)
+        getAllNewVideos()
+    }
+
+
+
+    const handleRatingChange = async (newRating) => {
```

Comment:
I would maybe call this function `changeRating` or `pushNewRating` rather than `handleRatingChange` - this function is the thing that _actually changes the rating on the server_ - "handle" as a prefix for a function name generally indicates something that happens _in response to_ the rest of the name (i.e. I would assume that this function is called _after the rating is changed in response to it_ rather than _actually changing the rating itself_).